google-cloud-storage/test/helper.rb

```diff
@@ -96,10 +101,16 @@ def random_bucket_hash name: random_bucket_name,
       "enableObjectRetention" => enable_object_retention,
       "softDeletePolicy" => soft_delete_policy_object(retention_duration_seconds: retention_duration_seconds),
       "hierarchicalNamespace" => hierarchical_namespace
-    }.delete_if { |_, v| v.nil? }
+    }
+    if soft_deleted
+      soft_delete_policy = soft_delete_policy_object retention_duration_seconds: retention_duration_seconds
```

Comment:
This method constructs hash for a bucket, not object. 
Can we have a separate method for getting soft delete policy for bucket ?